util/misc.go

```diff
@@ -559,6 +559,8 @@ var (
 	internalClientInit sync.Once
 	internalHTTPClient *http.Client
 	internalHTTPSchema string
+	// internalHTTPDefaultTimeOut is the default time out for http request
+	internalHTTPDefaultTimeOut = 10 * time.Second
```

Comment:
Can we make it as a config ? if it isn't set, we use a default value.

Reply:
> Can we make it as a config ? if it isn't set, we use a default value.

Thanks, I will modify it later. 
Since it only affects the query retrieving inside TiDB, how about making it as a system variable?

Reply:
I think you can improve the base library first. you can add internal HTTP config into the ```util/misc.go```